docroot/modules/custom/mass_entity_usage/tests/src/ExistingSite/EntityUsageTest.php

```diff
@@ -0,0 +1,216 @@
+   */
+  private function createEntities($curated_list_state) {
+    // Create a file to upload.
+    $destination = 'public://file-001.txt';
+    $file = File::create([
+      'uri' => $destination,
+    ]);
+    $file->setPermanent();
+    $file->save();
+    $src = 'modules/custom/mass_entity_usage/tests/fixtures/file-001.txt';
+
+    /** @var \Drupal\Core\File\FileSystemInterface $file_system */
+    $file_system = \Drupal::service('file_system');
+    $file_system->copy($src, $destination, TRUE);
+
+    // Create media item.
+    $media = $this->createMedia([
+      'title' => 'Llama',
+      'bundle' => 'document',
+      'field_upload_file' => [
+        'target_id' => $file->id(),
+      ],
+      'status' => 1,
+    ]);
+
+    $node_org = $this->createNode([
+      'type' => 'org_page',
+      'title' => 'Test Organization',
+      'uid' => $this->user->id(),
+      'moderation_state' => MassModeration::PUBLISHED,
+    ]);
+
+    // Entity usage tracking is done at the end of the requests, so we need to
+    // create the node through requests to updated the entity usage tracking
+    // table, instead of using $node->create.
+    // See patch related to issue #3015287 in composer.json.
+    $this->visit('/node/add/curated_list');
+    $this->getCurrentPage()->fillField('Title', 'Test Curated List');
+    $this->getCurrentPage()->fillField('Short title', 'Test Curated List Short Title');
+    $this->getCurrentPage()->fillField('Short description', 'Test Curated List Short Description');
+    $this->getCurrentPage()->fillField('Parent page', 'Test Organization (' . $node_org->id() . ') - Organization');
+    $this->getCurrentPage()->fillField('Organization(s)', 'Test Organization (' . $node_org->id() . ') - Organization');
+    $this->getCurrentPage()->fillField('Overview', $media->toLink()->toString());
+    $this->getCurrentPage()->fillField('Save as', $curated_list_state);
+    $this->getCurrentPage()->pressButton('Save');
+    $this->getCurrentPage()->hasContent('Curated List Test Curated List has been created.');
+
+    // Get last created node.
+    $res = \Drupal::entityQuery('node')->sort('nid', 'DESC')->range(0, 1)->execute();
+    $nid = reset($res);
+    $node_curated_list = Node::load($nid);
+
+    return [$media, $node_org, $node_curated_list];
+  }
+
+  /**
+   * Empties entity usage related queues.
+   */
+  private function emptyEntityUsageQueues() {
+    $this->clearQueue('entity_usage_tracker');
+    $this->clearQueue('entity_usage_regenerate_queue');
+    $this->clearQueue('mass_entity_usage_delete_not_current_revision_usages');
+  }
+
+  /**
+   * Process entity usage related queues.
+   */
+  private function processEntityUsageQueues() {
+    $this->runQueue('entity_usage_tracker');
+    $this->runQueue('mass_entity_usage_delete_not_current_revision_usages');
+  }
+
+  /**
+   * Assert that usage records are tracked properly.
+   */
+  public function testEntityUsageTracking() {
```

Comment:
We should test that we dont track UNPUBLISHED nodes. Also our paragraph handling is complex. Lets test that a bit deeper.